cpp/tests/groupby/bitwise_tests.cpp

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/aggregation.hpp>
+#include <cudf/groupby.hpp>
+
+auto bitwise_aggregate(cudf::column_view const& keys,
+                       cudf::column_view const& values,
+                       cudf::bitwise_op bit_op)
+{
+  std::vector<cudf::groupby::aggregation_request> requests;
+  requests.emplace_back();
+  requests[0].values = values;
+  requests[0].aggregations.emplace_back(
+    cudf::make_bitwise_aggregation<cudf::groupby_aggregation>(bit_op));
+  auto gb_obj = cudf::groupby::groupby(cudf::table_view({keys}));
+  auto result = gb_obj.aggregate(requests);
+  return std::pair{std::move(result.first->release().front()),
+                   std::move(result.second.front().results.front())};
+}
+
+template <typename T>
+using column = cudf::test::fixed_width_column_wrapper<T>;
+
+using keys_column = cudf::test::fixed_width_column_wrapper<int32_t>;
+
+class GroupByBitwiseTest : public cudf::test::BaseFixture {};
+
+template <typename T>
+class GroupByBitwiseTypedTest : public GroupByBitwiseTest {};
+
+// Test types for bitwise operations - only integer types
+using GroupByBitwiseTypedTestTypes =
+  cudf::test::Types<int8_t, int16_t, int32_t, int64_t, uint8_t, uint16_t, uint32_t, uint64_t>;
```

Comment:
There is a defined type for integers in `cudf::test` https://github.com/rapidsai/cudf/blob/d74d802f80b306f2681bdfd0ad838e25034001f7/cpp/include/cudf_test/type_lists.hpp#L190
Note that `std::integer` types includes `bool` so you may want to test for that too by using https://github.com/rapidsai/cudf/blob/d74d802f80b306f2681bdfd0ad838e25034001f7/cpp/include/cudf_test/type_lists.hpp#L195

Reply:
Note that `std::integer` types includes `bool` so it would be good to include tests for it.
If `bool` is not intended to be supported, then the https://github.com/rapidsai/cudf/pull/18551/files#diff-3b99730080ce6377b3ffa7517e2605a970ccc688355574258b5854fe5639a52bR270 would need to be modified to exclude `bool` I think. Though I would also think `bool` types would be a good candidate for bitwise aggregations.